tools/trinity.py

```diff
@@ -62,12 +63,13 @@ def execute(self,
                 outFasta,
                 min_contig_length=300,
                 JVMmemory=None,
-                threads=1):    # pylint: disable=W0221
+                threads=None):    # pylint: disable=W0221
         if JVMmemory is None:
             JVMmemory = self.jvm_mem_default
         outdir = tempfile.mkdtemp(prefix='trinity-')
-        if int(threads) < 1:
-            threads = 1
+        if not threads:
+            threads = 10000000
+        threads = min(threads, util.misc.available_cpu_count())
```

Comment:
Should this be a util function?

Reply:
I mostly copy-pasted this all over, but now that it's everywhere, I agree, it ought to be consolidated into a util.misc function.